### PR TITLE
Adding opentracing to log_anomaly_detector

### PR DESCRIPTION
**Pull request by zmhassan**
# Description
Adding visibility to make ml processes more visible

### CODE REVIEW

**Comment by zmhassan** [comment]
tracing

**Comment by zmhassan** [comment]
To test this out run:

`docker run --rm -p5775:5775/udp -p6831:6831/udp -p16686:16686 jaegertracing/all-in-one:latest`

**Comment by zmhassan** [comment]
To run ml job:

python app.py run --config-yaml config_files/.env_local_dir.yaml --single-run True

File: anomaly_detector/anomaly_detector_facade.py [hunk 1]
```diff
@@ -5,15 +5,33 @@
...
 import time
+from jaeger_client import Config
+from opentracing_instrumentation.request_context import get_current_span, span_in_context
 
 
 class AnomalyDetectorFacade:
     """For external interface for integration different adapters for custom models and training logic."""
 
-    def __init__(self, config, feedback_strategy=None):
+    def init_tracer(self, service):
```
Comment (HumairAK) [inline anomaly_detector/anomaly_detector_facade.py]: The convention is to have __init__ as the first class method, any particular reason why we have init_trace as the first method here?

Comment (zmhassan) [reply anomaly_detector/anomaly_detector_facade.py]: ohh. thank you for catching that. Its fixed now.

**Pull request merged**